ddtrace/tracer.py

```diff
@@ -43,83 +44,113 @@ def _parse_dogstatsd_url(url):
-                hostname = url_parsed.hostname
-                port = url_parsed.port
-                https = url_parsed.scheme == 'https'
-                # FIXME This is needed because of the way of configure() works right now, where it considers `port=None`
-                # to be "no port set so let's use the default".
-                # It should go away when we remove configure()
-                if port is None:
-                    if https:
-                        port = 443
-                    else:
-                        port = 80
-            elif url_parsed.scheme == 'unix':
-                uds_path = url_parsed.path
-            else:
-                raise ValueError('Unknown scheme `%s` for agent URL' % url_parsed.scheme)
+    enabled = attr.ib(default=True)
+    url = attr.ib(factory=_tracer_url)
+    dogstatsd_url = attr.ib(factory=_dogstatsd_url)
+    log = attr.ib(init=False, default=log, repr=False)
+    sampler = attr.ib(factory=AllSampler)
+    tags = attr.ib(factory=dict)
+    priority_sampler = attr.ib(factory=RateByServiceSampler)
+    collect_metrics = attr.ib(default=False)
+    filters = attr.ib(default=None)
+    # a buffer for service info so we don't perpetually send the same things
+    _context_provider = attr.ib(factory=DefaultContextProvider)
+    _wrap_executor = attr.ib(default=None)
+
+    # Internal
+    _pid = attr.ib(factory=getpid, init=False)
+    _services = attr.ib(init=False, factory=set, repr=False)
+    _runtime_worker = attr.ib(default=None, init=False, repr=False)
+
+    def __attrs_post_init__(self):
+        url_parsed = compat.parse.urlparse(self.url)
+        if url_parsed.scheme in ('http', 'https'):
+            hostname = url_parsed.hostname
+            port = url_parsed.port
+            https = url_parsed.scheme == 'https'
+            uds_path = None
+            # FIXME This is needed because of the way of configure() works right now, where it considers `port=None`
+            # to be "no port set so let's use the default".
+            # It should go away when we remove configure()
+            if port is None:
+                if https:
+                    port = 443
+                else:
+                    port = 80
+        elif url_parsed.scheme == 'unix':
+            hostname = 'localhost'
+            port = 8126
+            https = False
+            uds_path = url_parsed.path
+        else:
+            raise ValueError('Unknown scheme `%s` for agent URL' % url_parsed.scheme)
 
         # Apply the default configuration
         self.configure(
-            enabled=True,
             hostname=hostname,
             port=port,
             https=https,
             uds_path=uds_path,
-            sampler=AllSampler(),
-            context_provider=DefaultContextProvider(),
-            dogstatsd_url=dogstatsd_url,
+            dogstatsd_url=self.dogstatsd_url,
+            collect_metrics=self.collect_metrics,
+            settings={FILTERS_KEY: self.filters}
```

Comment:
Am I mistaken to conclude the only value of `settings` we handle is one where we look for the value for `FILTERS_KEY`?

Reply:
No, you're right. It seems somebody had an idea to use settings for other things but that never happened.

I'm not a fan of having a magic dict with undocumented key passed to object constructors in general, so I adapted and pulled the filter out.